framework/include/cppmicroservices/Any.h

```diff
@@ -452,6 +452,31 @@ class BadAnyCastException : public std::bad_cast
   std::string _msg;
 };
 
+
+namespace detail
+{
+US_Framework_EXPORT std::string GetDemangledName(const std::type_info& typeInfo);
+}
+
+/**
+ * \ingroup gr_any
+ *
+ * A utility function used to throw a BadAnyCastException object
+ * containing an exception message containing the source and target type names.
+ *
+ * \param funcName The throwing function's name.
+ * \param anyTypeName A string representing the Any object's underlying type.
+ * \throws cppmicroservices::BadAnyCastException
+ */
+template <typename TargetType>
+void ThrowBadAnyCastException(const std::string& funcName, const std::string& anyTypeName)
```

Comment:
I think this should go into the `detail` namespace as well and not be part of the public API docs. Passing both the target and source std::type_info objects as arguments to `ThrowBadAnyCastException` would also allow us to hide its implementation in a cpp file as well as not having to export `GetDemangledName`.